crates/extensions/c8y_mapper_ext/src/compatibility_adapter.rs

```diff
@@ -140,16 +149,17 @@ fn convert_to_old_agent_request(
 }
 
 fn convert_from_old_agent_response(
+    prefix: &str,
     cmd_type: &str,
     payload: &[u8],
 ) -> Result<Option<MqttMessage>, String> {
     if let Ok(Value::Object(response)) = serde_json::from_slice(payload) {
         if let Some(Value::String(cmd_id)) = response.get("id") {
             // The new mapper expects command ids with a specific prefix
-            let topic_name = if cmd_id.starts_with("c8y-mapper") {
+            let topic_name = if cmd_id.starts_with(prefix) {
                 format!("te/device/main///cmd/{cmd_type}/{cmd_id}")
             } else {
-                format!("te/device/main///cmd/{cmd_type}/c8y-mapper-{cmd_id}")
+                format!("te/device/main///cmd/{cmd_type}/{prefix}-{cmd_id}")
```

Comment:
question: so as of now we generate commands in this converter using a mapper topic prefix, but do we process (in `OperationHandler`) only commands matching the prefix?

Reply:
I've added a change to fix this